### PR TITLE
Skip known relval failures in ppc64le for test-os jobs

### PR DESCRIPTION
**Pull request by aandvalenzuela**
@smuzaffar I followed the same pattern that we are currently using for aarch64, but I can also explicitly grep for the failing tests:
```
23234.0_TTbar_14TeV+2026D94 | Step1-FAILED
24834.0_TTbar_14TeV+2026D98 | Step1-FAILED
24834.911_TTbar_14TeV+2026D98_DD4hep | Step1-FAILED
24896.0_CloseByPGun_CE_E_Front_120um+2026D98 | Step1-FAILED
24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98 | Step1-FAILED
25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR | Step2-FAILED
```

### CODE REVIEW

File: bin/test-cmssw.sh [hunk 1]
```diff
@@ -2,7 +2,11 @@
 CMSREP="cmsrep.cern.ch"
 ADD_PKGS=""
 RUN_TESTS="false"
-TEST_OK_MATCH="tests passed, 0 1 0 0 0 0 0 0 0 0 failed"
+if [ $(echo ${SCRAM_ARCH} | grep '_aarch64_' | wc -l) -eq 1 ] ; then
+  TEST_OK_MATCH="tests passed, 0 1 0 0 0 0 0 0 0 0 failed"
+elif [ $(echo ${SCRAM_ARCH} | grep '_ppc64le_' | wc -l) -eq 1 ] ; then
+  TEST_OK_MATCH="tests passed, 0 5 1 0 0 0 0 0 0 0 0 failed"
+fi
```
[the remark this answers is not in the document]
Comment (smuzaffar) [reply bin/test-cmssw.sh:9]: ormay be better to get the failing workflows from opensearch

[the remark this answers is not in the document]
Comment (aandvalenzuela) [reply bin/test-cmssw.sh:9]: Thanks! Working on it

**Comment by aandvalenzuela** [comment]
I have tested it with jobs https://cmssdt.cern.ch/jenkins/job/test-os-el8/ and https://cmssdt.cern.ch/jenkins/view/All/job/cms-containers-run-cmssw-test/

See sample output for known errors at https://cmssdt.cern.ch/jenkins/job/test-os-el8/1469/console: 
```
+ cat /scratch/cmsbuild/jenkins_a/workspace/test-os-el8/res.txt
el8_ppc64le_gcc11
el8_ppc64le_gcc11.BOOT.OK
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.BUILD.OK
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.23234.0.Step1.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.24834.0.Step1.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.24834.911.Step1.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.24896.0.Step1.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.24900.0.Step1.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.RELVAL.25034.999.Step2.KNOWN
el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-15-2300.TEST.OK
++ grep '\.ERR$' /scratch/cmsbuild/jenkins_a/workspace/test-os-el8/res.txt
++ wc -l
+ '[' 0 -gt 0 ']'
+ echo ALL_OK
ALL_OK
```

I goes along with https://github.com/cms-sw/cms-bot/pull/2040 to get the known relval errors from OpenSearch.

File: bin/test-cmssw.sh [hunk 2]
```diff
@@ -49,9 +47,24 @@ run_the_matrix () {
     if grep ' tests passed' matrix.log ; then
       if [ $(grep ' tests passed' matrix.log | sed 's|.*tests passed||' | tr ' ' '\n' | grep '^[1-9]' | wc -l) -eq 0 ] ; then
         RES="OK"
-      elif [ $(echo ${SCRAM_ARCH} | grep '_aarch64_' | wc -l) -eq 1 ] ; then
-        if [ $(grep "${TEST_OK_MATCH}" matrix.log | wc -l) -eq 1 ] ; then
-          RES="OK"
+      else
+        echo "Checking known errors..."
+        if [ -f "$WORKSPACE/relval-$SCRAM_ARCH" ]; then
```
Comment (smuzaffar) [inline bin/test-cmssw.sh:52]: it is better to pass `$WORKSPACE/relval-$SCRAM_ARCH` as command-line argument.

**Comment by smuzaffar** [comment]
this looks good now. tested https://cmssdt.cern.ch/jenkins/job/test-os-el8/1492/console and now it properly matches the wf,step and exit code.
```
18:23:21 + echo 'Found Relval 25034.999:step3:62720'
18:23:21 Found Relval 25034.999:step3:62720
18:23:21 + grep -q '^WF:25034.999:step3:62720$' ib-relvals.txt
18:23:21 + echo 'Known error'
....
18:24:13 el8_ppc64le_gcc11
18:24:13 el8_ppc64le_gcc11.BOOT.OK
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.BUILD.OK
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.23234.0.2.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.24834.0.2.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.24834.911.2.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.24896.0.2.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.24900.0.2.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.RELVAL.25034.999.3.KNOWN
18:24:13 el8_ppc64le_gcc11.CMSSW_13_3_X_2023-08-22-2300.TEST.OK
```

**Pull request merged**